packages/trader/src/App/Containers/RealAccountSignup/personal-details.jsx

```diff
@@ -0,0 +1,235 @@
+                                        'dc-input__trailing-icon',
+                                        props.trailing_icon.props.className,
+                                    ),
+                                },
+                            )
+                        }
+                        <label className='dc-input__label' htmlFor={field.id}>
+                            {props.label || props.placeholder}
+                        </label>
+                        <ErrorMessage name={field.name}>
+                            {
+                                (msg) => (
+                                    <p className='dc-input__error'>
+                                        {msg}
+                                    </p>
+                                )
+                            }
+                        </ErrorMessage>
+                    </div>
+                )
+            }
+        </Field>
+    );
+};
+
+class PersonalDetails extends Component {
+    validatePersonalDetails = (values) => {
+        const validations = {
+            first_name: [
+                v => !!v,
+                v => v.length > 2 && v.length < 50,
+            ],
+            last_name: [
+                v => !!v,
+                v => v.length > 2 && v.length < 50,
+            ],
+        };
+
+        const mappedKey = {
+            first_name: localize('First name'),
+            last_name : localize('Last name'),
+        };
+        const errors    = {};
+
+        Object.entries(validations)
+            .forEach(([key, rules]) => {
+                if (rules.some(v => !v(values[key]))) {
+                    errors[key] = localize(`${mappedKey[key]} is required`);
+                }
+            });
+
+        return errors;
+    };
+
+    render() {
+        return (
+            <Formik
+                initialValues={{
+                    first_name   : this.props.value.first_name,
+                    last_name    : this.props.value.last_name,
+                    date_of_birth: this.props.value.date_of_birth || toMoment().subtract(18, 'years').unix(),
+                }}
+                validate={this.validatePersonalDetails}
+                onSubmit={(values, actions) => {
+                    this.props.onSubmit(this.props.index, values, actions.setSubmitting);
+                }}
+            >
+                {
+                    ({
+                        handleSubmit,
+                        isSubmitting,
+                        errors,
+                        values,
+                        touched,
+                    }) => (
+                        <form onSubmit={handleSubmit}>
+                            <div className='personal-details-form'>
+                                <Input
+                                    name='first_name'
+                                    label={localize('First name')}
+                                    placeholder={localize('John')}
```

Comment:
does first_name and last name need `error={touched.first_name && errors.first_name}`?

Reply:
No, it is being passed with `{...fields}`.